threescale/http/client.go

```diff
@@ -299,6 +313,37 @@ func (c *Client) handleRateLimitExtensions(resp *http.Response) *api.RateLimits
 	return rl
 }
 
+func (c *Client) handleNoBodyExtensionForAuth(resp *http.Response, extensions api.Extensions) *threescale.AuthorizeResult {
+	var rl *api.RateLimits
+	if _, ok := extensions[api.LimitExtension]; ok {
+		rl = c.handleRateLimitExtensions(resp)
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		return &threescale.AuthorizeResult{
+			Authorized:  true,
+			RawResponse: resp,
+			AuthorizeExtensions: threescale.AuthorizeExtensions{
+				RateLimits: rl,
+			},
+		}
+	}
+
+	return &threescale.AuthorizeResult{
+		Authorized:  false,
+		ErrorCode:   c.parseRejectionReasonHeader(resp),
```

Comment:
I think this currently requires an extension to enable.

Reply:
that's ok, if it empty so be it, nothing will change. its more expensive to check if the extension was set and parse it than just do a lookup on the header directly, wdyt?